github_server/action.yml

```diff
@@ -1,53 +1,143 @@
 name: "ALKiln/github_server: Isolated docassemble tests"
-description: "Automatically test docassemble interviews on GitHub's servers, in an isolated environment. Avoids docassemble reload errors and other external factors"
+description: "Download and start up a temporary docassemble server on GitHub's servers, in an isolated environment. Avoids docassemble reload errors and other external factors."
 
+# Bug? Race condition? Exit with 52 when waiting for container to finish starting
+# https://github.com/SuffolkLITLab/ALKiln/actions/runs/5912900256/job/16036751264#step:4:188
+# Had no completion messages, but it exited with an error anyway
+# Took 121 seconds
+# The rerun of the test completed that step just fine, though it took less time
+# Took 120 and was fine: https://github.com/SuffolkLITLab/ALKiln/actions/runs/5913167036/job/16037288791
+# 150 secs with exit code 56:
+# https://github.com/SuffolkLITLab/ALKiln/actions/runs/5914677262/job/16040233419#step:4:196
+# 120 sec with exit code 56:
+# https://github.com/SuffolkLITLab/ALKiln/actions/runs/5914963614/job/16040773624#step:4:206
```

Comment:
Looked into this a bit more and I was incorrect on our call; exit code [56 comes from `curl`](https://everything.curl.dev/usingcurl/returns), and is a "Failure when receiving data from the peer". Error 52 is apparently [empty reply from the server](https://stackoverflow.com/a/42837622/11416267). Both of these are probably expected, because DA is starting servers and everything, so if it starts the nginx server in the middle of our call, `curl` would get this error.

The confusing thing here was that by default, GitHub runs `bash` steps as `shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}`. `-e` means that if [any command in the bash script fails, then the entire step fails](https://stackoverflow.com/a/9952249/11416267). Since `curl` returns lots of non-zero exit codes, we don't want the step to fail anytime it happens, we just want to ignore it (if it fails, we will still print out the curl response in the logs for debugging, so that should be fine).

GitHub's [suggested way of avoiding these extra args to bash](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference) is to use `shell: bash {0}`, which I recommend we do in this case.

Reply:
What about our own failure that we're adding in? Won't that be eaten too? Maybe I can just do a try/catch on the curl.

Reply:
Running without `-e` doesn't eat all exits, just those that come from commands; `exit` will always exit as normal. Running without `-e` is closer the C standard library, where a call to a function can fail, as long as it's not a null error or other segfault, the program will continue. See [the bash docs](https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin) for more info there, but I'm happy to explain more if needed.

Bash doesn't have a try-catch in the way you're thinking of. Thinking about it that way, `-e` causes failing commands to throw exceptions, and so removing it just stops that behavior.